google-cloud-policy_troubleshooter-v1/lib/google/cloud/policy_troubleshooter/v1/iam_checker/client.rb

```diff
@@ -0,0 +1,375 @@
+              # the gRPC module only when it's required.
+              # See https://github.com/googleapis/toolkit/issues/446
+              require "gapic/grpc"
+              require "google/cloud/policytroubleshooter/v1/checker_services_pb"
+
+              # Create the configuration object
+              @config = Configuration.new Client.configure
+
+              # Yield the configuration if needed
+              yield @config if block_given?
+
+              # Create credentials
+              credentials = @config.credentials
+              credentials ||= Credentials.default scope: @config.scope
+              if credentials.is_a?(String) || credentials.is_a?(Hash)
+                credentials = Credentials.new credentials, scope: @config.scope
+              end
+              @quota_project_id = @config.quota_project
+              @quota_project_id ||= credentials.quota_project_id if credentials.respond_to? :quota_project_id
+
+              @iam_checker_stub = ::Gapic::ServiceStub.new(
+                ::Google::Cloud::PolicyTroubleshooter::V1::IamChecker::Stub,
+                credentials:  credentials,
+                endpoint:     @config.endpoint,
+                channel_args: @config.channel_args,
+                interceptors: @config.interceptors
+              )
+            end
+
+            # Service calls
+
+            ##
+            # Checks whether a member has a specific permission for a specific resource,
+            # and explains why the member does or does not have that permission.
+            #
+            # @overload troubleshoot_iam_policy(request, options = nil)
+            #   Pass arguments to `troubleshoot_iam_policy` via a request object, either of type
+            #   {::Google::Cloud::PolicyTroubleshooter::V1::TroubleshootIamPolicyRequest} or an equivalent Hash.
+            #
+            #   @param request [::Google::Cloud::PolicyTroubleshooter::V1::TroubleshootIamPolicyRequest, ::Hash]
+            #     A request object representing the call parameters. Required. To specify no
+            #     parameters, or to keep all the default parameter values, pass an empty Hash.
+            #   @param options [::Gapic::CallOptions, ::Hash]
+            #     Overrides the default settings for this call, e.g, timeout, retries, etc. Optional.
+            #
+            # @overload troubleshoot_iam_policy(access_tuple: nil)
+            #   Pass arguments to `troubleshoot_iam_policy` via keyword arguments. Note that at
+            #   least one keyword argument is required. To specify no parameters, or to keep all
+            #   the default parameter values, pass an empty Hash as a request object (see above).
+            #
+            #   @param access_tuple [::Google::Cloud::PolicyTroubleshooter::V1::AccessTuple, ::Hash]
+            #     The information to use for checking whether a member has a permission for a
+            #     resource.
```

Comment:
It's too bad the fields of this single parameter weren't flattened into the request, but I guess that's the API design. Maybe they'll add additional params in the future?